### PR TITLE
Release v0.28.0: Module package separation for command implementations

### LINKED ISSUE #199: [Change] Module package separation for command implementations
## Why

The modular command registry exists, but command implementations still live in `src/specfact_cli/commands/`. Module packages are mostly re-export shells, which keeps command ownership centralized and increases merge conflict risk.

## What Changes

- Move command implementations into module-local `src/commands.py` files under `src/specfact_cli/modules/<name>/src/`.
- Update each module `src/app.py` to import `app` from local `commands`.
- Keep backward-compatible re-export shims in `src/specfact_cli/commands/`.
- Execute migration in tiers (small to heavyweight) with verification gates.
- Update architecture and contributor docs that reference old command implementation locations.

## Acceptance Criteria

- Each migrated module has local command implementation and local app wiring.
- Legacy imports from `specfact_cli.commands.*` continue to work via shims.
- Tier-by-tier tests, contract checks, and representative CLI help checks pass.
- Documentation reflects module-local command implementation model.

*OpenSpec Change Proposal: arch-02-module-package-separation*

### PR DESCRIPTION
**Pull request by djm81**
# Description

Release v0.28.0 consolidating all changes from `dev` into `main`. This release delivers the **module package separation** architecture (OpenSpec change `arch-02-module-package-separation`), completing the second phase of the CLI modular architecture initiative.

**New Features** #199

**Contract References**: Module boundary import contracts, compatibility shim validation.

## Type of Change

Please check all that apply:

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Contract enforcement (adding/updating `@icontract` decorators)
- [x] 🧪 Test enhancement (scenario tests, property-based tests)
- [x] 🔧 Refactoring (code improvement without functionality change)

### Added

- **Module package separation for command implementations** (OpenSpec change `arch-02-module-package-separation`, fixes #199)
  - **Module-local command sources**: command implementations now live under `src/specfact_cli/modules/<module>/src/commands.py` and module app entrypoints import from local command modules.
  - **Boundary regression checks**: added tests to prevent new non-`app` dependencies from `specfact_cli.commands.*` and protect module encapsulation.
  - **Shared helper extraction**: common cross-command helpers moved to stable shared utilities to reduce coupling between module command packages.

### Changed

- **Compatibility shims for migration**: legacy `src/specfact_cli/commands/*.py` files are now compatibility shims focused on `app` export, preserving CLI behavior during transition while reducing symbol-level coupling.
- **Version**: Bumped to 0.28.0 (minor: architectural feature/refactor, backward compatible).

## Contract-First Testing Evidence

**Required for all changes affecting CLI commands or public APIs:**

### Contract Validation

- [x] **Runtime contracts added/updated** (`@icontract` decorators on public APIs)
- [x] **Type checking enforced** (`@beartype` decorators applied)
- [x] **CrossHair exploration** completed: `hatch run contract-test-exploration`
- [x] **Contract violations** reviewed and addressed

### Test Execution

- [x] **Contract validation**: `hatch run contract-test-contracts` ✅
- [x] **Contract exploration**: `hatch run contract-test-exploration` ✅
- [x] **Scenario tests**: `hatch run contract-test-scenarios` ✅
- [x] **Full test suite**: `hatch run contract-test-full` ✅

### Test Quality

- [x] **CLI commands tested** with typer test client
- [x] **Edge cases covered** with Hypothesis property tests
- [x] **Error handling tested** with invalid inputs
- [x] **Rich console output verified** manually or with snapshots

## How Has This Been Tested?

**Contract-First Approach**: Module boundary import tests validate that `specfact_cli.commands.*` shims only re-export `app`, and module `commands.py` files contain the full implementations. Migration compatibility tests ensure all command groups load correctly through the new module paths.

### Manual Testing

- [x] Tested CLI commands manually
- [x] Verified rich console output
- [x] Tested with different input scenarios
- [x] Checked error messages for clarity

### Automated Testing

- [x] Contract validation passes
- [x] Property-based tests cover edge cases
- [x] Scenario tests cover user workflows
- [x] All existing tests still pass

### Test Environment

- Python version: 3.11, 3.12, 3.13
- OS: Ubuntu 22.04 (CI), macOS 14

## Checklist

- [x] My code follows the style guidelines (PEP 8, ruff format, isort)
- [x] I have performed a self-review of my code
- [x] I have added/updated contracts (`@icontract`, `@beartype`)
- [x] I have added/updated docstrings (Google style)
- [x] I have made corresponding changes to documentation
- [x] My changes generate no new warnings (basedpyright, ruff, pylint)
- [x] All tests pass locally
- [x] I have added tests that prove my fix/feature works
- [x] Any dependent changes have been merged

## Quality Gates Status

- [x] **Type checking** ✅ (`hatch run type-check`)
- [x] **Linting** ✅ (`hatch run lint`)
- [x] **Contract validation** ✅ (`hatch run contract-test-contracts`)
- [x] **Contract exploration** ✅ (`hatch run contract-test-exploration`)
- [x] **Scenario tests** ✅ (`hatch run contract-test-scenarios`)

## Screenshots/Recordings (if applicable)

N/A - architectural refactoring with no UI changes.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/specfact_cli/modules/backlog/src/commands.py [hunk 1]
```diff
@@ -0,0 +1,2800 @@
...
+            resources_path = find_package_resources_path("specfact_cli", "resources/templates/backlog")
+            built_in_loaded = False
+            if resources_path and resources_path.exists():
+                registry.load_templates_from_directory(resources_path)
+                built_in_loaded = True
+            else:
+                # Fallback: Try relative to repo root (development mode)
+                repo_root = Path(__file__).parent.parent.parent.parent
+                resources_templates_dir = repo_root / "resources" / "templates" / "backlog"
+                if resources_templates_dir.exists():
```
[the remark this answers is not in the document]
Comment (djm81) [reply src/specfact_cli/modules/backlog/src/commands.py]: Fix applied.

File: src/specfact_cli/modules/contract/src/commands.py [hunk 2]
```diff
@@ -0,0 +1,1244 @@
...
+
+    # Create new contract index entry
+    endpoints_count = 0
+    checksum = None
+    if contract_file.exists():
+        try:
+            contract_data = load_openapi_contract(contract_file)
+            endpoints_count = count_endpoints(contract_data)
+            checksum = hashlib.sha256(contract_file.read_bytes()).hexdigest()
+        except Exception:
```
[the remark this answers is not in the document]
Comment (djm81) [reply src/specfact_cli/modules/contract/src/commands.py]: Will be applied.
